js/pdfkit.es5.js

```diff
@@ -4973,24 +4973,60 @@ var ICCProfile = /*#__PURE__*/function () {
 
       if (buffers.length == 0) return;
       return Buffer.concat(buffers);
-    }
+    } // A PNG file is composed of an 8-byte PNG signature followed by a sequence of well-structured chunks.
+    // PNG chunks are laid out like so:
+    // [ Length (4 bytes) | Type (4 bytes) | Data (length bytes) | CRC (4 bytes) ]
+    // And the data in the iCCP chunk is laid out like so:
+    // [ Profile name (1-79 bytes) | Null terminator (1 byte) | Compression method (1 byte) | Compressed profile (n bytes)]
+    // References:
+    // - https://www.w3.org/TR/2003/REC-PNG-20031110/#5PNG-file-signature
+    // - http://www.libpng.org/pub/png/spec/1.2/PNG-Chunks.html#C.iCCP
+
   }, {
     key: "extractFromPNG",
-    value: function extractFromPNG(png) {
-      var pos = png.indexOf('iCCP');
+    value: function extractFromPNG(pngBuffer) {
+      var offset = 8; // Skip PNG signature
 
-      if (pos === -1) {
-        console.log('No iCCP chunk found');
-        return;
+      while (offset < pngBuffer.length) {
+        var chunkLength = pngBuffer.readUInt32BE(offset);
+        var chunkType = pngBuffer.slice(offset + 4, offset + 8).toString('ascii');
+
+        if (chunkType === 'IEND') {
+          break; // End of PNG file
+        }
+
+        if (chunkType === 'iCCP') {
+          var dataStart = offset + 8;
+          var dataEnd = dataStart + chunkLength;
+          var data = pngBuffer.slice(dataStart, dataEnd); // Find null separator and compression method
+
+          var nullPos = data.indexOf(0);
+
+          if (nullPos === -1 || nullPos === data.length - 1) {
+            throw new Error('Invalid iCCP chunk format');
+          } // const profileName = data.slice(0, nullPos).toString('ascii');
+
+
+          var compressionMethod = data[nullPos + 1];
+
+          if (compressionMethod !== 0) {
+            throw new Error('Unsupported compression method');
+          }
+
+          var compressedProfile = data.slice(nullPos + 2);
+
+          try {
+            var iccProfile = zlib.inflateSync(compressedProfile);
+            return iccProfile;
+          } catch (error) {
+            throw new Error('Failed to decompress ICC profile: ' + error.message);
+          }
+        }
+
+        offset += chunkLength + 12; // Length + Type + Data + CRC
```

Comment:
How did you get 12 for this? You have this comment: `[ Length (4 bytes) | Type (4 bytes) | Data (length bytes) | CRC (4 bytes) ]`. What does `Data (length bytes)` mean?

Reply:
so, `Length` + `Type` + `CRC` is 4 + 4 + 4 = 12. Then `chunkLength` = is the "length bytes" piece. We store the chunk data length in the Length header, which is 4 bytes long. That length header is a number that represents the byte count of the data block. so "length bytes" is "<length> bytes", like ... N bytes. does... that make sense?

Reply:
ah, yes. that makes sense.